packages/server/src/fhir/operations/agentpush.ts

```diff
@@ -57,8 +67,49 @@ export const agentPushHandler = asyncWrap(async (req: Request, res: Response) =>
     return;
   }
 
-  await getRedis().publish(getReferenceString(agent), JSON.stringify({ type: 'push', remote: device.url, ...message }));
-  sendOutcome(res, allOk);
+  const message: AgentTransmitRequest = {
+    type: 'agent:transmit:request',
+    remote: device.url,
+    contentType: params.contentType,
+    body: params.body,
+  };
+
+  // If not waiting for a response, publish and return
+  if (!params.waitForResponse) {
+    await publishMessage(agent, message);
+    sendOutcome(res, allOk);
+    return;
+  }
+
+  // Otherwise, open a new redis connection in "subscribe" state
+  // message.callback = getReferenceString(agent) + '-' + randomUUID();
+  message.callback = randomUUID();
+
+  const redisSubscriber = getRedis().duplicate();
+  await redisSubscriber.subscribe(message.callback);
```

Comment:
Cool, this is a pretty interesting way to use subscriptions. I'm curious what the overhead of duplicating / subscribing to a new topic in a high throughput HL7 session, but I think this probably works for how it's currently being used.

Reply:
Yeah, agree.  At minimum we should probably use a more sane channel name than `randomUUID`.

ChatGPT claims that Redis can handle this performantly.... I guess we'll see?

Reply:
Yeah I can believe it.

Maybe one cool thing would be to set up a granular profiling system for testing particular parts / flows of the system that is easy to expand as the codebase expands. That would be super cool. Not sure what that would look like yet but I'll be thinking about it